File: BackEnd/src/main/java/hector/vico/EntreBicis/entity/PermisUsuari.java
package hector.vico.EntreBicis.entity;

/**
 * Enum que defineix els tipus de permisos o rols que pot tenir un usuari dins del sistema EntreBicis.
 *
 * <ul>
 *   <li><b>ADMIN:</b> Usuari amb permisos d’administració completa (gestió de rutes, usuaris, recompenses, etc.).</li>
 *   <li><b>CICLISTA:</b> Usuari normal que pot registrar rutes, consultar recompenses, etc.</li>
 * </ul>
 *
 * Aquest enum es pot utilitzar, per exemple, per controlar l'accés a funcionalitats o vistes específiques.
 */
public enum PermisUsuari {
    ADMIN,    // Permís d’administrador del sistema
    CICLISTA  // Permís d’usuari ciclista normal
}

File: FrontEnd/app/src/main/java/cat/copernic/hvico/entrebicis/Core/Model/PermisUsuari.kt
package cat.copernic.hvico.entrebicis.Core.Model

/**
 * Representa els diferents rols que pot tenir un usuari dins del sistema.
 *
 * @property CICLISTA Usuari amb permisos limitats, pot realitzar rutes, veure recompenses i modificar el seu perfil.
 * @property ADMIN Usuari amb permisos complets, pot gestionar usuaris, recompenses i paràmetres del sistema.
 */
enum class PermisUsuari {
    CICLISTA,
    ADMIN
}
